crates/biome_js_analyze/src/lint/nursery/use_explicit_type.rs

```diff
@@ -740,3 +719,105 @@ fn is_type_assertion(syntax: &SyntaxNode<JsLanguage>) -> bool {
         }
     })
 }
+
+fn handle_any_function(func: &AnyJsFunction) -> Option<TextRange> {
+    if func.return_type_annotation().is_some() {
+        return None;
+    }
+
+    if is_direct_const_assertion_in_arrow_functions(func) {
+        return None;
+    }
+
+    if is_iife(func) {
+        return None;
+    }
+
+    if is_function_used_in_argument_or_array(func) {
+        return None;
+    }
+
+    if is_arrow_func(func) && (can_inline_function(func) || is_function_inside_typed_return(func)) {
+        return None;
+    }
+
+    if is_higher_order_function(func) {
+        return None;
+    }
+
+    if is_typed_function_expressions(func) {
+        return None;
+    }
+
+    let func_range = func.syntax().text_trimmed_range();
+    if let Ok(Some(AnyJsBinding::JsIdentifierBinding(id))) = func.id() {
+        return Some(TextRange::new(
+            func_range.start(),
+            id.syntax().text_trimmed_range().end(),
+        ));
+    }
+
+    Some(func_range)
+}
+
+/// Checks if a variable declarator needs to have an explicit type.
+fn handle_variable_declarator(
+    declarator: &JsVariableDeclarator,
+    model: &SemanticModel,
+) -> Option<TextRange> {
+    let variable_declaration = declarator
+        .parent::<JsVariableDeclaratorList>()?
+        .parent::<JsVariableDeclaration>()?;
+    let is_top_level = declarator
+        .syntax()
+        .ancestors()
+        .find_map(JsVariableStatement::cast)
+        .is_some_and(|statement| {
+            statement
+                .syntax()
+                .parent()
+                .is_some_and(|parent| JsModuleItemList::can_cast(parent.kind()))
+        })
+        || variable_declaration
+            .parent::<JsVariableDeclarationClause>()
+            .is_some();
+
+    let identifier = declarator.id().ok()?;
+    let identifier = identifier.as_any_js_binding()?.as_js_identifier_binding()?;
+    let is_exported = model.is_exported(identifier);
+
+    if !is_top_level || !is_exported {
```

Comment:
I wonder if we should cover all variable declaration at top-level or namespace level. What do you think?
```suggestion
    if !is_top_level {
```